tests/devices/qutrit_mixed/test_qutrit_mixed_apply_operation.py

```diff
@@ -0,0 +1,307 @@
+        """Test a snapshot is recorded property when provided a tag"""
+        state = request.getfixturevalue(state)
+        initial_state = math.asarray(state, like=ml_framework)
+
+        debugger = self.Debugger()
+        tag = "dense"
+        new_state = apply_operation(
+            qml.Snapshot(tag), initial_state, debugger=debugger, is_state_batched=len(shape) != 2
+        )
+
+        assert new_state.shape == initial_state.shape
+        assert math.allclose(new_state, initial_state)
+
+        assert list(debugger.snapshots.keys()) == [tag]
+        assert debugger.snapshots[tag].shape == shape
+        assert math.allclose(debugger.snapshots[tag], math.reshape(initial_state, shape))
+
+    def test_snapshot_with_measurement(self, ml_framework, state, shape, request):
+        """Test a snapshot with measurement throws NotImplementedError"""
+        state = request.getfixturevalue(state)
+        initial_state = math.asarray(state, like=ml_framework)
+
+        debugger = self.Debugger()
+        with pytest.raises(NotImplementedError):
+            _ = apply_operation(
+                qml.Snapshot(measurement=qml.expval(qml.GellMann(0, 1))),
+                initial_state,
+                debugger=debugger,
+                is_state_batched=len(shape) != 2,
+            )
+
+
+@pytest.mark.parametrize("ml_framework", ml_frameworks_list)
+class TestBroadcasting:  # pylint: disable=too-few-public-methods
+    """Tests that broadcasted operations (not channels) are applied correctly."""
+
+    broadcasted_ops = [
+        qml.TRX(np.array([np.pi, np.pi / 2]), wires=2, subspace=(0, 1)),
+        qml.TRY(np.array([np.pi, np.pi / 2]), wires=2, subspace=(0, 1)),
+        qml.TRZ(np.array([np.pi, np.pi / 2]), wires=2, subspace=(1, 2)),
+        qml.QutritUnitary(
+            np.array([unitary_group.rvs(27), unitary_group.rvs(27)]),
+            wires=[0, 1, 2],
+        ),
+    ]
+    unbroadcasted_ops = [
+        qml.THadamard(wires=2),
+        qml.TClock(wires=2),
+        qml.TShift(wires=2),
+        qml.TAdd(wires=[1, 2]),
+        qml.TRX(np.pi / 3, wires=2, subspace=(0, 2)),
+        qml.TRY(2 * np.pi / 3, wires=2, subspace=(1, 2)),
+        qml.TRZ(np.pi / 6, wires=2, subspace=(0, 1)),
+        qml.QutritUnitary(unitary_group.rvs(27), wires=[0, 1, 2]),
+    ]
+    num_qutrits = 3
+    num_batched = 2
+    dims = (3**num_qutrits, 3**num_qutrits)
+
+    @pytest.mark.parametrize("op", broadcasted_ops)
+    def test_broadcasted_op(self, op, ml_framework, three_qutrit_state):
+        """Tests that batched operations are applied correctly to an unbatched state."""
+
+        state = three_qutrit_state
+        flattened_state = state.reshape(self.dims)
+
+        res = apply_operation(op, qml.math.asarray(state, like=ml_framework))
+        missing_wires = 3 - len(op.wires)
+        mat = op.matrix()
+        expanded_mats = [
+            np.kron(np.eye(3**missing_wires), mat[i]) if missing_wires else mat[i]
```

Comment:
It looks like this is testing only cases with 3 qubits where the operation is on the last wire? Recommend to add a sanity check for a different cases, like operations on the second wire, etc. You can also use `qml.matrix` with a `wire_order` argument to construct the matrices in a way that incorporates additional qubits.